.github/workflows/ci-workflow.yml

```diff
@@ -21,12 +29,32 @@ jobs:
       - name: Upload app binary
         uses: actions/upload-artifact@v2
         with:
-          name: boilerplate-app-debug
+          name: boilerplate-app-nanoS-debug
+          path: bin
+
+  job_nanoX_build_debug:
+    name: Build debug app for NanoX
+    runs-on: ubuntu-latest
+
+    container:
+      image: ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder:latest
+
+    steps:
+      - name: Clone
+        uses: actions/checkout@v2
+
+      - name: Build
+        run: |
+          make DEBUG=1 BOLOS_SDK=$NANOX_SDK
+
+      - name: Upload app binary
+        uses: actions/upload-artifact@v2
+        with:
+          name: boilerplate-app-nanoX-debug
           path: bin
 
   job_scan_build:
     name: Clang Static Analyzer
-    needs: job_build_debug
```

Comment:
Out of curiosity, why did you delete this requirement?
With this line, job is not run if project fails to compile. Is it a bad pattern?

Reply:
Fair point. There's two concurrent goals on this: 

- first one (old behavior) gives more meaning to each job: if `job_scan_build` fails, it means the analyzer failed - not the compilation which was checked on the previous `job_build_debug` step.
- second one (current behavior) aims faster feedbacks. `job_scan_build` takes **A**, and `job_build_debug` **B**, this CI gives feedback in **max(A, B)** whereas old one takes **A+B**. But `job_scan_build` failure could mean a `job_build_debug` failure.

Phrased like that makes the second behavior sound petty, so I'm reverting to the old behavior.